libflux/src/core/parser/mod.rs

```diff
@@ -577,6 +577,57 @@ impl Parser {
             MonoType::Invalid
         }
     }
+
+    #[cfg(test)]
+    fn parse_record(&mut self) -> Record {
+        //( "{" [Properties] "}" ) | ( "{" Tvar "with" Properties "}" )
+        let _ = self.expect(TOK_LBRACE);
+        // can be (Properties) or (Tvar "with" Properties)
+        let t = self.peek();
+        if t.lit.to_uppercase() != (t.lit) {
+            let properties = self.parse_properties();
+            let record = Record {
+                base: self.base_node_from_token(&t),
```

Comment:
I think you'll need to use `base_node_from_tokens` here and pass in the tokens for the right and left braces.